core/primitives/src/network.rs

```diff
@@ -9,13 +10,29 @@ use near_crypto::{KeyType, PublicKey, SecretKey, Signature};
 use crate::hash::{hash, CryptoHash};
 use crate::types::{AccountId, EpochId};
 
+/// Peer id is the public key.
+#[derive(
+    BorshSerialize, BorshDeserialize, Clone, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
+)]
+pub struct PeerId(Arc<PeerIdInner>);
+
 /// Peer id is the public key.
 #[derive(
     BorshSerialize, BorshDeserialize, Clone, Eq, PartialOrd, Ord, Serialize, Deserialize, Hash,
 )]
-pub struct PeerId(pub PublicKey);
+pub struct PeerIdInner(pub PublicKey);
 
 impl PeerId {
+    pub fn new(key: PublicKey) -> Self {
+        Self(Arc::new(PeerIdInner(key)))
+    }
+
+    pub fn public_key(&self) -> PublicKey {
+        self.0 .0.clone()
```

Comment:
space before a dot (or is it on purpose?)

Reply:
In older rusts, `self.0.0.clone()` is be lexed as `self. 0.0 .clone`, that is, `0.0` would be treated as a floating point literal by the compiler. So adding a space used to be an idiom for this. Modern rust has a hack in the compiler to handle this in intuitive way, but I guess rustfmt still adds a space to be on a safe side.

Reply:
Fixed